### ISSUE TITLE
🐛 [BUG]

### ISSUE THREAD

**Issue by hsingll** [mas-cli/mas#412]
## Your Environment

- `mas version`:stable 1.8.3
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 10.15.7 (19H1323)

### mas Install Method

- [X] `brew install mas` (homebrew-core)
- [] [mas-cli/tap](https://github.com/mas-cli/homebrew-tap)
- [] `.pkg` installer from [releases](https://github.com/mas-cli/mas/releases)
- [] Built from source
  - Fork/branch: ? (e.g. mas-cli/main)
  - Xcode version: 10.?

## Describe the Bug

> A clear and concise description of what the bug is.

mas list command output disalignment.

## To Reproduce

1. install software above.

## Expected Behavior

> A clear and concise description of what you expected to happen.

## Actual Behavior

> A clear and concise description of what actually happened.

## Screenshots, Terminal Output

> If applicable, add screenshots to help explain your problem.

<img width="448" alt="Snipaste_2021-09-09_10-30-09" src="https://user-images.githubusercontent.com/26346635/132612976-94e38c11-52fe-4a76-bbc6-b9cab679ae27.png">

```bash
$ mas list
...
```

## Additional Context

> Add any other context about the problem here.
> Did it work in a previous version?

**Comment by drewauff** [comment]
This is much worse when piping through fzf. `column -t` "fixes" this a little but everything is so spread out.

**Comment by rgoldberg** [comment]
### TL;DR: solution:
`mas` should be a shell script that calls a Swift executable that outputs only JSON (let's call it `mas-json` for now).

All formatting of output should be done by the `mas` script.

Users can configure the output environment variables, a config file, or both.

If `mas` doesn't (yet) offer certain output options, users can edit the `mas` script directly.

e.g., when `mas list` is called, it will run something like the following (the real script would use an environment variable with a default if it isn't set, but I'm keeping the example simple):

```zsh
mas-json list | jq -r '(.[] | [.id, .name, .version]) | @tsv' | column -ts $'\t' -R 1
```

`mas-json list` outputs the JSON for the installed apps to stdout.

`jq -r '(.[] | [.id, .name, .version]) | @tsv'` outputs the id, name & version of each element of the JSON array (i.e. of each installed app) to stdout as a tab separated line.

`column -ts $'\t' -R 1` right justifies the first column (id), and creates a properly spaced table from the tab-separated output of `jq`.

The mas brew formula would depend on `jq` & `util-linux`, because `jq` doesn't come with macOS & `column` from macOS doesn't support right justification (at least the versions I've seen).

### Reasons why the above solution is advisable:
It is basically impossible to determine the displayed or occupied widths of characters in Swift because:
- no existing library to do so that I've found (iTerm 2 has some code, but it probably only applies to iTerm 2, and probably isn't always correct)
- mas Swift code doesn't have access to the terminal itself, it just outputs to stdout, while the `column` command has access to the terminal, so it can easily space out columns.
- different terminal apps might have completely different outputs
- different versions of the same terminal app might have completely different outputs
- different fonts used by a terminal app might have completely different outputs
- various combining characters interact with other characters in weird ways
- carious characters have different displayed & occupied widths (if displayed is wider than occupied, the wide character is superimposed over subsequent characters), so you need to keep track of both values
- various terminals have bugs in how they handle various characters
- much more complexity

UNIX principle: Do one thing very well / separate concerns; in this case, i.e. separate data from display. Users could completely change the display however they want. It's much easier to parse JSON than text that was attempted to be formatted into columns. Also, the JSON can include many fields that aren't displayed by default, so users can choose or remove columns easily, apply text colors or other formatting, etc.

**Comment by phatblat** [comment]
My worry about this project providing a user-editable `mas` script front-end is how would Homebrew install this script? I don't know if a formula can install a script that isn't overwritten on every update. If the new `mas` script is customized by users but then overwritten by every update, that would undermine the value of being able to customize the script.

I'm not opposed to the idea of a new front-end script, just calling out a potential snag that needs to be investigated.

Many CLI tools have an optional JSON output by passing a `--json` flag. Homebrew even has two versions of [its JSON output format](https://formulae.brew.sh/docs/api/). This type of dual output formatting (see #190) is what I've figured `mas` would do eventually, but have never had the time to implement it myself.

**Comment by rgoldberg** [comment]
I wasn't planning on having users normally edit the script. It was more if they wanted to edit it, then they could, then submit changes to the repo.

But, we should be able to offer enough config options via environment variables, a config file, or just allowing them to pipe JSON to `jq` or to some other command to format output however they see fit.
